xbmc/video/Teletext.cpp

```diff
@@ -32,21 +32,20 @@
 #include "filesystem/SpecialProtocol.h"
 #include "guilib/GraphicContext.h"
 
-#define SDL_memset4(dst, val, len)		\
-do {						\
-	uint32_t _count = (len);		\
-	uint32_t _n = (_count + 3) / 4;		\
-	uint32_t *_p = static_cast<uint32_t *>(dst);	\
-	uint32_t _val = (val);			\
-	if (len == 0) break;			\
-        switch (_count % 4) {			\
-        case 0: do {    *_p++ = _val;		\
-        case 3:         *_p++ = _val;		\
-        case 2:         *_p++ = _val;		\
-        case 1:         *_p++ = _val;		\
-		} while ( --_n );		\
-	}					\
-} while(0)
+static inline void SDL_memset4(void* dst, uint32_t val, size_t len)
+{
+  size_t _n = (len + 3) / 4;
+  uint32_t *_p = static_cast<uint32_t *>(dst);
+  if (len == 0)
+    return;
+  switch (len % 4) {
+    case 0: do { *_p++ = val;
```

Comment:
Thanks garbear. Leave loop unrolling to the compiler I say. This code looks horrible